File: packages/rag/src/vector-store.ts
/**
 * Vector store interface: index and search by embedding.
 */
import type { DocumentChunk, VectorDocument, SearchResult } from "./types.js";

export interface VectorStore {
  /** Add documents with embeddings */
  add(documents: VectorDocument[]): Promise<void>;
  /** Search by embedding; returns top K results */
  search(embedding: number[], topK: number, filter?: Record<string, unknown>): Promise<SearchResult[]>;
  /** Delete by document IDs */
  delete?(ids: string[]): Promise<void>;
}

File: python/src/agent_framework/rag/vector_store.py
"""Vector store interface."""

from abc import ABC, abstractmethod
from agent_framework.rag.types import SearchResult, VectorDocument


class VectorStore(ABC):
    """Index and search by embedding."""

    @abstractmethod
    async def add(self, documents: list[VectorDocument]) -> None:
        ...

    @abstractmethod
    async def search(
        self,
        embedding: list[float],
        top_k: int,
        filter_: dict | None = None,
    ) -> list[SearchResult]:
        ...
